pkg/minikube/node/cache.go

```diff
@@ -151,14 +146,8 @@ func beginDownloadKicBaseImage(g *errgroup.Group, cc *config.ClusterConfig, down
 			}
 
 			if driver.IsDocker(cc.Driver) {
-				if image.ExistsImageInDaemon(img) {
-					klog.Infof("%s exists in daemon, skipping pull", img)
-					finalImg = img
-					return nil
-				}
-
 				klog.Infof("Downloading %s to local daemon", img)
-				err = image.WriteImageToDaemon(img)
```

Comment:
how about only move the exact "Download" part of the logic into the download image and still keep the `WriteImageToDaemon` in the `image` package ? 

feel free to break down the WriteImageToDaemon into smaller part if the "download" part of it is not modular.

Reply:
That would be ideal, but the trouble is the lock. In order to properly lock the download, it needs to download and write while locked. Separating the download and the write would mean we'd need a function anyway to wrap both those using the lock (and moving the tests while we're at it) which just moves the problem. The download would also be returning an unresolved image. This moves the downloading behavior out of the download function (as in, the function wouldn't be blocked while downloading), which doesn't match the semantics of the rest of the package.

@medyagh We could move just the image writing stuff back to the image package and have the download functions directly call those writing functions. Thoughts? Or is that just unnecessary?